### PR TITLE
SAAS-6116: Add Page 2 and CYA for repair and maintenance

### PR DESCRIPTION
**Pull request by lukastymo**

### CODE REVIEW

File: app/views/journeys/expenses/repairsandmaintenance/RepairsAndMaintenanceAmountView.scala.html [hunk 1]
```diff
@@ -35,39 +35,35 @@
         businessId: BusinessId,
         accountingType: AccountingType)(implicit request: Request[_], messages: Messages)
 
-@layout(pageTitle = title(form, messages("repairsAndMaintenanceAmount.title"))) {
-    
+@layout(pageTitle = title(form, messages(s"repairsAndMaintenanceAmount.heading.$userType"))) {
```
Comment (Jake-Raffe) [inline app/views/journeys/expenses/repairsandmaintenance/RepairsAndMaintenanceAmountView.scala.html:38]: Page title should now be '.heading.individual'

File: app/views/components/SingleAmountContent.scala.html [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            field = form("value"),
+            label = LabelViewModel(
+              HtmlContent(content)
+            )
+        )
+        .asNumeric()
+        .withWidth(Fixed10)
+        .withPrefix(PrefixOrSuffix(content = Text("£")))
+    )
+</div>
```
Comment (Jake-Raffe) [inline app/views/components/SingleAmountContent.scala.html:36]: EOF error

File: conf/messages.cy [hunk 3]
```diff
@@ -679,34 +679,37 @@ totalExpensesCYA.heading = totalExpensesCYA
...
+repairsAndMaintenanceAmount.heading.individual = How much did you spend on repairs and maintenance?
+repairsAndMaintenanceAmount.heading.agent = How much did your client spend on repairs and maintenance?
 repairsAndMaintenanceAmount.details = Understanding repairs and maintenance expenses
-repairsAndMaintenanceAmount.checkYourAnswersLabel = RepairsAndMaintenanceAmountCheckYourAnswersLabel
-repairsAndMaintenanceAmount.error.nonNumeric = Enter your repairsAndMaintenanceAmount using numbers
-repairsAndMaintenanceAmount.error.required = Enter your repairsAndMaintenanceAmount
-repairsAndMaintenanceAmount.error.wholeNumber = Enter your repairsAndMaintenanceAmount using whole numbers
-repairsAndMaintenanceAmount.error.outOfRange = RepairsAndMaintenanceAmount must be between {0} and {1}
+repairsAndMaintenanceAmount.checkYourAnswersLabel.individual = How much did you spend on repairs and maintenance?
+repairsAndMaintenanceAmount.checkYourAnswersLabel.agent = How much did your client spend on repairs and maintenance?
```
Comment (Jake-Raffe) [inline conf/messages.cy:687]: We don't need these if they are the same as the headings. This should have failed the MessagesSpec test.

File: conf/messages.cy [hunk 4]
```diff
@@ -679,34 +679,37 @@ totalExpensesCYA.heading = totalExpensesCYA
...
+repairsAndMaintenanceAmount.error.required.individual = Enter the amount you spent on repairs and maintenance
+repairsAndMaintenanceAmount.error.required.agent = Enter the amount your client spent on repairs and maintenance
+repairsAndMaintenanceAmount.error.nonNumeric.individual = The amount you spent on repairs and maintenance must be a number
+repairsAndMaintenanceAmount.error.nonNumeric.agent = The amount your client spent on repairs and maintenance must be a number
+repairsAndMaintenanceAmount.error.overMax.individual = The amount you spent on repairs and maintenance must be less than £100,000,000,000
+repairsAndMaintenanceAmount.error.overMax.agent = The amount your client spent on repairs and maintenance must be less than  £100,000,000,000
+repairsAndMaintenanceAmount.error.lessThanZero.individual = The amount you spent on repairs and maintenance must be more than £0
+repairsAndMaintenanceAmount.error.lessThanZero.agent = The amount your client spent on repairs and maintenance must be more than £0
 repairsAndMaintenanceAmount.change.hidden = RepairsAndMaintenanceAmountChangeHidden
```
Comment (Jake-Raffe) [inline conf/messages.cy:696]: This needs to be a sentence for screen readers e.g. 'Change how much was spent on repairs and maintenance' (keep it user neutral)

File: test/controllers/journeys/expenses/repairsandmaintenance/RepairsAndMaintenanceAmountControllerSpec.scala [hunk 5]
```diff
@@ -16,45 +16,38 @@
...
 
         val result = route(application, getRequest).value
 
         status(result) mustBe OK
+
+        val view = application.injector.instanceOf[RepairsAndMaintenanceAmountView]
+        val expectedView =
+          view(form, mode, userType, currTaxYear, stubBusinessId, accountingType)(getRequest, messages(application, Language.English)).toString()
         contentAsString(result) mustEqual expectedView
       }
```
Comment (Jake-Raffe) [inline test/controllers/journeys/expenses/repairsandmaintenance/RepairsAndMaintenanceAmountControllerSpec.scala:52]: This needs a lot more tests if we're not using the BigDecimalBaseSpec

File: app/views/journeys/expenses/repairsandmaintenance/RepairsAndMaintenanceAmountView.scala.html (hunk 1 above)
Comment (lukastymo) [reply app/views/journeys/expenses/repairsandmaintenance/RepairsAndMaintenanceAmountView.scala.html:38]: done

File: app/views/components/SingleAmountContent.scala.html (hunk 2 above)
Comment (lukastymo) [reply app/views/components/SingleAmountContent.scala.html]: done

File: conf/messages.cy (hunk 4 above)
Comment (lukastymo) [reply conf/messages.cy:696]: nice, I was wonder how we use it and forgot to ask

File: test/base/BigDecimalGetAndPostQuestionBaseSpec.scala [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+) extends ControllerSpec {
+  val onPageLoadRoute: String
+  val onSubmitRoute: String
+
+  def createForm(userType: UserType): Form[BigDecimal]
+
+  def expectedView(expectedForm: Form[_], scenario: TestScenario)(implicit request: Request[_], messages: Messages, application: Application): String
+
+  val validAnswer: BigDecimal        = 100.00
+  val filledUserAnswers: UserAnswers = UserAnswers(userAnswersId).set(page, validAnswer).success.value
```
Comment (Jake-Raffe) [inline test/base/BigDecimalGetAndPostQuestionBaseSpec.scala]: this needs .set(page, validAnswer, Some(businessId))

File: conf/messages.cy (hunk 3 above)
Comment (lukastymo) [reply conf/messages.cy:687]: MessageSpec was green

**Review by Jake-Raffe — DISMISSED** [review]

**Review by Jake-Raffe — APPROVED** [review]

**Pull request merged**